### PR TITLE
feat: allow deletion of roles

### LINKED ISSUE #5881: Removed event organizer still receives email when participants sign up
**Describe the bug**
As describe on the tittle. I was added as an organizer of one event for testing, then the owner removed my organizer access. I could not access the organizing page anymore. However I still received email every time when someone bought a ticket for that event. 

**To Reproduce**
Test event I mentioned https://eventyay.com/e/3dbaaa50

**Expected behaviour**
Removed organizer should not receive ticket purchasing email. 

**Desktop**
- Ubuntu 18.04

### PR DESCRIPTION
**Pull request by sachinchauhan2889**
Fixes #5881 

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: app/templates/components/events/view/overview/manage-roles.hbs [hunk 1]
```diff
@@ -65,6 +55,27 @@
           {{t 'No records to show'}}
         </div>
       {{/each}}
+      {{/if}}
+      {{#if (eq this.roleType 'accepted')}}
```
Comment (iamareebjamal) [inline app/templates/components/events/view/overview/manage-roles.hbs:59]: Why this check?

File: app/templates/components/events/view/overview/manage-roles.hbs [hunk 2]
```diff
@@ -65,6 +55,27 @@
...
+      {{#each this.data.usersEventsRoles as |eventRole|}}
+        <tr>
+          <td>{{eventRole.user.email}}</td>
+          <td>{{eventRole.role.titleName}}</td>
+          {{#if (and @editable  (not-eq eventRole.role.titleName 'Owner'))}}
```
Comment (iamareebjamal) [inline app/templates/components/events/view/overview/manage-roles.hbs:64]: Check name, not title name

File: app/templates/components/events/view/overview/manage-roles.hbs [hunk 3]
```diff
@@ -19,43 +19,33 @@
...
         <th>{{t 'Role'}}</th>
-        {{#if (and @editable  (eq this.roleType 'pending'))}}
+        {{#if @editable}}
           <th>{{t 'Options'}}</th>
         {{/if}}
       </tr>
     </thead>
     <tbody>
+      {{#if (eq this.roleType 'pending')}}
       {{#each this.roleInvites as |roleInvite|}}
```
Comment (iamareebjamal) [inline app/templates/components/events/view/overview/manage-roles.hbs:29]: indentation

File: app/templates/components/events/view/overview/manage-roles.hbs [hunk 4]
```diff
@@ -65,6 +55,26 @@
           {{t 'No records to show'}}
         </div>
       {{/each}}
+      {{else}}
+      {{#each this.data.usersEventsRoles as |eventRole|}}
```
Comment (iamareebjamal) [inline app/templates/components/events/view/overview/manage-roles.hbs:59]: indentation

**Comment by maze-runnar** [comment]
I am getting error on deletion - 
{"errors": [{"status": 500, "source": null, "title": "Unknown error", "detail": "Unknown error"}], "jsonapi": {"version": "1.0"}}
Screenshot from 2021-01-14 17-33-46

File: app/templates/components/events/view/overview/manage-roles.hbs [hunk 5]
```diff
@@ -1,13 +1,13 @@
...
-    </LinkTo>
-  {{/if}}
+    {{#if @editable}}
+      <button class="ui blue button ml-auto" {{action 'openAddUserRoleModal'}}>{{t 'Add People'}}</button>
+    {{/if}}
+    {{#if @showLink}}
+      <LinkTo @route="events.view.team.index" class="item ml-auto">
+        <button class="ui blue button">{{t 'Edit'}}</button>
+      </LinkTo>
+    {{/if}}
```
Comment (iamareebjamal) [inline app/templates/components/events/view/overview/manage-roles.hbs:10]:
```suggestion
  {{#if @editable}}
    <button class="ui blue button ml-auto" {{action 'openAddUserRoleModal'}}>{{t 'Add People'}}</button>
  {{/if}}
  {{#if @showLink}}
    <LinkTo @route="events.view.team.index" class="item ml-auto">
      <button class="ui blue button">{{t 'Edit'}}</button>
    </LinkTo>
  {{/if}}
```

**Pull request merged**